arbor/morph/em_morphology.cpp

```diff
@@ -40,6 +40,9 @@ em_morphology::em_morphology(const morphology& m):
         auto idx = util::make_range(morph_.branch_indexes(i));
         branch_lengths_.push_back(dist2root_[idx.back()]- dist2root_[idx.front()]);
     }
+    if (morph_.spherical_root()) {
```

Comment:
I am treating it as the distance from the center, where `0,0` is the centre of the sphere, not an edge.

Reply:
Yup, but it's inconsistent: the branch length of a non-spherical root is indeed the full length of the cable. We have other mechanisms for getting the radius of the root node, and the branch length should consistently return the length of the branch.

Reply:
The spherical root is a special case, and I feel uncomfortable with this approach because it attempts to "orient" the branch.
You can also construct inconsistent interpretations from this, e.g. the distance between the sample 0 and sample 1, which is at the start of the next branch, is now `length(branch 0)/2`.

Reply:
On the second point first, the samples define the geometry, not the other way around: there is nothing inconsistent about the distance between the sample 0 and sample 1 not being the length of the branch, if the branch is a sphere.

Spherical somata *are* oriented: one end has all the other branches sticking out of it, and the other end doesn't. Those branches share a common end point, just as they would if they were attached to a non-spherical cable root with the same topology.

Reply:
But none of that changes the fact that geometrically, the length of a cable shaped like a sphere should be its diameter, not its radius. If I were to substitute the spherical root with an equivalent cable, the length should not change.